File: CS-Examples/05_Images/AlignPictureWithinCell.cs
using System;
using System.Data.OleDb;
using System.Drawing;
using System.Collections;
using System.ComponentModel;
using System.Windows.Forms;
using System.Data;
using Spire.Xls;
using Spire.Xls.Core.Spreadsheet;

namespace AlignPictureWithinCell
{
	public partial class Form1 : Form
	{
        public Form1()
        {
            InitializeComponent();
        }

		private void btnRun_Click(object sender, System.EventArgs e)
		{
            // Create a new workbook
            Workbook workbook = new Workbook();

            // Get the first worksheet in the workbook
            Worksheet sheet = workbook.Worksheets[0];

            // Set the text in cell A1 as "Align Picture Within A Cell:"
            sheet.Range["A1"].Text = "Align Picture Within A Cell:";

            // Set the vertical alignment of cell A1 to top
            sheet.Range["A1"].Style.VerticalAlignment = VerticalAlignType.Top;

            // Insert an image at the specific cell (1, 1)
            string picPath = @"..\..\..\..\..\..\Data\SpireXls.png";
            ExcelPicture picture = sheet.Pictures.Add(1, 1, picPath);

            // Adjust the column width and row height so that the cell can contain the picture
            sheet.Columns[0].ColumnWidth = 40;
            sheet.Rows[0].RowHeight = 200;

            // Set the horizontal offset of the image within the cell to 100
            picture.LeftColumnOffset = 100;

            // Set the vertical offset of the image within the cell to 25
            picture.TopRowOffset = 25;

            // Specify the name of the resulting Excel file
            string result = "Result-AlignPictureWithinCell.xlsx";

            // Save the workbook to the specified file using Excel 2013 format
            workbook.SaveToFile(result, ExcelVersion.Version2013);

            // Dispose of the workbook object to release resources
            workbook.Dispose();

            // Launch the file.
            ExcelDocViewer(result);
		}

        private void ExcelDocViewer(string fileName)
        {
            try
            {
                System.Diagnostics.Process.Start(fileName);
            }
            catch { }
        }

        private void btnClose_Click(object sender, EventArgs e)
        {
            Close();
        }
	}
}

File: VB-Examples/05_Images/AlignPictureWithinCell.vb
Imports System.Data.OleDb
Imports System.Collections
Imports System.ComponentModel
Imports Spire.Xls
Imports Spire.Xls.Core.Spreadsheet

Namespace AlignPictureWithinCell
	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
		End Sub

		Private Sub btnRun_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnRun.Click
            'Create a new instance of the Workbook class.
            Dim workbook As New Workbook()

            'Assign the first worksheet to the "sheet" variable.
            Dim sheet As Worksheet = workbook.Worksheets(0)

            'Set the text content of cell A1 to "Align Picture Within A Cell:".
            sheet.Range("A1").Text = "Align Picture Within A Cell:"
            'Sets the vertical alignment of the cell A1 to the top.
            sheet.Range("A1").Style.VerticalAlignment = VerticalAlignType.Top

            'Specifie the file path of the image.
            Dim picPath As String = "..\..\..\..\..\..\Data\SpireXls.png"
            'Adds an image to the specified cell (1, 1).
            Dim picture As ExcelPicture = sheet.Pictures.Add(1, 1, picPath)

            'Set the width of the first column to 40.
            sheet.Columns(0).ColumnWidth = 40
            'Sets the height of the first row to 200.
            sheet.Rows(0).RowHeight = 200

            'Set the horizontal offset of the image.
            picture.LeftColumnOffset = 100
            'Sets the vertical offset of the image.
            picture.TopRowOffset = 25

            'Specify the file name for the resulting workbook.
            Dim result As String = "Result-AlignPictureWithinCell.xlsx"

            'Save the workbook to a file with the specified name and version.
            workbook.SaveToFile(result, ExcelVersion.Version2013)
            ' Release the resources used by the workbook
            workbook.Dispose()

            'Launch the MS Excel file.
            ExcelDocViewer(result)
		End Sub

		Private Sub ExcelDocViewer(ByVal fileName As String)
			Try
				Process.Start(fileName)
			Catch
			End Try
		End Sub

		Private Sub btnClose_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnClose.Click
			Close()
		End Sub
	End Class
End Namespace
